src/macaron/repo_finder/repo_finder_deps_dev.py

```diff
@@ -164,11 +156,9 @@ def get_latest_version(purl: PackageURL) -> tuple[PackageURL | None, RepoFinderI
             RepoFinderInfo.FOUND_FROM_LATEST,
         )
 
-    def _create_urls(self, purl: PackageURL) -> tuple[list[str], RepoFinderInfo]:
-        """
-        Create the urls to search for the metadata relating to the passed artifact, and report on that process.
-
-        If a version is not specified, remote API calls will be used to try and find one.
+    @staticmethod
+    def get_attestation(purl: PackageURL) -> tuple[dict | None, bool]:
```

Comment:
Is this function intended to support extracting the attestation for all type of PURL or only PYPI one ? 
I think this function assumes that the link to the attestation provided by deps.dev (e.g. https://pypi.org/integrity/ultralytics/8.3.70/ultralytics-8.3.70-py3-none-any.whl/provenance) to a PYPI provenance.

Reply:
It is currently only intended for PyPI. I have moved the PyPI specific extraction code to the related registry, and will report unsupported for other types.